### PR TITLE
Laravel 5.3 Shift

### PR DESCRIPTION
**Pull request by mattstauffer**
This pull request includes the changes for upgrading to Laravel 5.3. Feel free to commit any additional changes to the `shift-3768` branch.

**Before merging**, you should:

- Checkout the `shift-3768` branch
- Review **all** pull request comments for additional changes
- Run `composer update` (if the pre-commands fail, add `--no-scripts`)
- Thoroughly test your application

If you would like more help with your upgrade, check out the [human services from Laravel Shift](https://laravelshift.com/human-services). And if you want to level-up your Git skills, check out [Getting Git](https://gettinggit.com).

### CODE REVIEW

**Comment by mattstauffer** [comment]
❌ Laravel 5.3 split the `Auth/AuthController.php` into the `Auth/LoginController.php` and the `Auth/RegisterController.php`. Since yours appears to be customized, Shift did not remove it. You should compare your `Auth/AuthController.php` with the newly added controllers and merge your changes accordingly.

**Comment by mattstauffer** [comment]
⚠ If you are using *Authentication*, you should upgrade any routes using `AuthController` and `PasswordController` to use the new authentication controllers. Laravel recommends using the `Auth::routes()` which registers the proper routes for the new authentication controllers. This method also registers a `POST` route for `/logout` instead of a `GET` route.

**Comment by mattstauffer** [comment]
❌ Shift could not upgrade `app/Providers/AuthServiceProvider.php`. You will need to compare yours against the [5.3 version](https://github.com/laravel-shift/laravel-5.3/blob/master/app/Providers/AuthServiceProvider.php) and merge the new changes.

**Comment by mattstauffer** [comment]
❌ Shift was not able to apply all the updates to your `app/Http/Kernel.php` file. You should compare yours with the [5.3 version](https://github.com/laravel-shift/laravel-5.3/blob/master/app/Http/Kernel.php) to ensure it is fully updated.

**Comment by mattstauffer** [comment]
ℹ Laravel 5.3 changed the redirect URL for an authenticated user from `/` to `/home` in `app/Http/Middleware/RedirectIfAuthenticated.php`. You may wish to change this depending on your app configuration. Otherwise, it may cause an infinite redirection loop.

**Comment by mattstauffer** [comment]
⚠ Laravel 5.3 now includes a top-level `routes` folder. In addition, it separates routes into *API*, *console*, and *web* files.Shift included a default `routes/api.php` and `routes/console.php` as well as migrated your `app/Http/routes.php` to `routes/web.php`. You should review your routes to see if they should be separated into one of the new route files.

**Comment by mattstauffer** [comment]
❌ Shift could not upgrade the following configuration files since they differed from the default Laravel version. You will need to compare these configuration files against the [Laravel 5.3 configuration files](https://github.com/laravel-shift/laravel-5.3/blob/master/config) and merge any changes:

- [ ] config/auth.php
- [ ] config/mail.php
- [ ] config/services.php

**Comment by mattstauffer** [comment]
❌ Laravel 5.3 has a `laravel/cashier` dependency of `~7.0`. Ensure it meets the new version requirement of `~7.0`.

**Comment by mattstauffer** [comment]
❌ Laravel 5.3 `where` performs a "loose" comparison. In addition, the *strict* parameter was removed. If you need to perform a "strict" comparison, you can use `whereStrict` instead.

You should review the following usages of `where` to ensure you do not require "strict" compaison:

- [ ] app/Jobs/NotifyUserOfNewGistComment.php
- [ ] app/Jobs/NotifyUserOfNewGistComments.php
- [ ] app/Http/Controllers/Auth/AuthController.php

**Comment by mattstauffer** [comment]
❌ The Query Builder returns collections, instead of plain arrays, in Laravel 5.3. You will need to upgrade your code to use collections or chain the `all()` method onto your query to return a plain array.

You should review the following usages of Query Builder:

- [ ] database/migrations/2015_08_17_132640_allow_blank_names.php

**Comment by mattstauffer** [comment]
ℹ If you are using *Queues*, you no longer need to specify the `--daemon` option when calling the `queue:work`. This is assumed in Laravel 5.3. Use the `--once` option to process a single job.

**Comment by mattstauffer** [comment]
ℹ If you are using the `database` driver for your *Queue*, you will need to update your `jobs` and `failed_jobs` table. Review the *Queue* section of the [Laravel 5.3 Upgrade Guide](https://laravel.com/docs/5.3/upgrade#upgrade-5.3.0) for more details and feel free to use the [migrations from Laravel Shift](https://github.com/laravel-shift/laravel-5.3/tree/master/database/migrations).

**Comment by mattstauffer** [comment]
ℹ Laravel 5.3 updated several of the default assets as well as adding support for [Vue.js](https://vuejs.org). If you are using Laravel to [compile assets](https://laravel.com/docs/5.3/elixir) you may want to review the latest versions of [`resources/assets/js/app.js`](https://github.com/laravel-shift/laravel-5.3/blob/master/resources/assets/js/app.js) and [`resources/assets/sass/app.scss`](https://github.com/laravel-shift/laravel-5.3/blob/master/resources/assets/sass/app.scss) and update your project accordingly.

**Comment by mattstauffer** [comment]
⚠ Laravel 5.3 added configuration for your application name in `config/app.php`. Be sure to set this value as it is used in notifications, like the password reset email.

File: composer.json [hunk 1]
```diff
@@ -8,10 +8,10 @@
     "license": "MIT",
     "type": "project",
     "require": {
-        "php": ">=5.5.9",
-        "laravel/framework": "5.2.*",
+        "php": ">=5.6.4",
+        "laravel/framework": "5.3.*",
         "laravel/socialite": "^2.0",
-        "laravel/cashier": "~5.0",
+        "laravel/cashier": "~7.0",
```
Comment (imjohnbon) [inline composer.json]: To my knowledge, cashier isn't even used anymore. Perhaps we can remove it as a dependency entirely?

Comment (mattstauffer) [inline composer.json]: @imjohnbon agreed

**Comment by imjohnbon** [comment]
@mattstauffer This should be good to review. All tests pass and I've confirmed it works locally 👍

**Pull request merged**